pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java

```diff
@@ -982,8 +982,7 @@ protected void internalAddProducer(Producer producer) throws BrokerServiceExcept
 
     private void tryOverwriteOldProducer(Producer oldProducer, Producer newProducer)
             throws BrokerServiceException {
-        if (newProducer.isSuccessorTo(oldProducer) && !isUserProvidedProducerName(oldProducer)
-                && !isUserProvidedProducerName(newProducer)) {
+        if (newProducer.isSuccessorTo(oldProducer)) {
```

Comment:
Why do you remove `isUserProvidedProducerName` check?

Reply:
Because `! isUserProvidedProducerName()` will prevent the new producer override the old one